### PR TITLE
Add use_external_data_format flag to transformer optimization pass

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/passes/onnx/transformer_optimization.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def _run_for_config(
         if config["optimization_options"]:
             self._set_fusion_options(run_config)
 
-        optimizer = transformers_optimizer.optimize_model(input=model.model_path, **run_config)
+        optimizer = transformers_optimizer.optimize_model(input=model.model_path, use_external_data_format=True, **run_config)
```
Comment (guotuofeng) [inline olive/passes/onnx/transformer_optimization.py:104]: need check ort version here since the flag doesn't exist with 1.15

Comment (guotuofeng) [reply olive/passes/onnx/transformer_optimization.py:104]: https://github.com/microsoft/Olive/blob/main/olive/systems/azureml/aml_pass_runner.py#L66-L76C1

need to be removed after upgrading to ORT 1.16

Comment (trajepl) [reply olive/passes/onnx/transformer_optimization.py:104]: Yes, maybe no hurry to add this changes now?

Comment (xiaoyu-work) [reply olive/passes/onnx/transformer_optimization.py:104]: @guotuofeng Thanks for reminding. Will add version check.

Comment (xiaoyu-work) [reply olive/passes/onnx/transformer_optimization.py:104]: @trajepl There are a couple of users who are trying large whisper models with multilingual with ORT nightly. We want to update Olive as well to support it. I can add a version check here!

Comment (guotuofeng) [reply olive/passes/onnx/transformer_optimization.py:104]: @xiaoyu-work , it seems the use_external_data_format flag is removed in PR https://github.com/microsoft/onnxruntime/pull/16912

Comment (trajepl) [reply olive/passes/onnx/transformer_optimization.py:104]: With latest nightly package, the transformer optimization will infer the flag of external data automatically.

**Comment by trajepl** [comment]
Should we close this PR? Because the bug is fixed in following PR where user does not need to provide the flag of `use_external_data_format`
https://github.com/microsoft/onnxruntime/pull/16912

**Comment by xiaoyu-work** [comment]
> Should we close this PR? Because the bug is fixed in following PR where user does not need to provide the flag of `use_external_data_format` [microsoft/onnxruntime#16912](https://github.com/microsoft/onnxruntime/pull/16912)

Yeah, closing.

**Pull request closed without merging**